sys/kern/uio.c

```diff
@@ -81,6 +81,21 @@ int uiomove(void *buf, size_t n, uio_t *uio) {
   return error;
 }
 
+void uio_save(const uio_t *uio, uiosave_t *save) {
+  save->us_resid = uio->uio_resid;
+  save->us_iovcnt = uio->uio_iovcnt;
+  save->us_iov_off = uio->uio_iov_off;
+}
+
+void uio_restore(uio_t *uio, const uiosave_t *save) {
+  size_t nbytes = save->us_resid - uio->uio_resid;
+  uio->uio_resid += nbytes;
+  uio->uio_offset -= nbytes;
+  uio->uio_iov_off = save->us_iov_off;
+  uio->uio_iov -= save->us_iovcnt - uio->uio_iovcnt;
+  uio->uio_iovcnt = save->us_iovcnt;
+}
```

Comment:
What do you think about `inline` for these functions?

Reply:
In order to be inlined, these functions would need to be defined in the header `uio.h`. I don't think it's worth it.